File: ncarg2d/src/libncarg/support/entsr.f
C
C $Id: entsr.f,v 1.7 2008-07-27 00:17:29 haley Exp $
C                                                                      
C                Copyright (C)  2000
C        University Corporation for Atmospheric Research
C                All Rights Reserved
C
C The use of this Software is governed by a License Agreement.
C
      SUBROUTINE ENTSR (IROLD,IRNEW)
C
C This routine returns, in IROLD, the current value of the recovery
C flag.  If the value of IRNEW is non-zero, ENTSR also resets the
C recovery flag: IRNEW = 1 activates recovery; IRNEW = 2 deactivates
C recovery.
C
C The common blocks SECOMI and SECOMC are used to hold shared variables
C of types INTEGER and CHARACTER, respectively, for the routine SETER
C and associated routines.  For descriptions of these variables and for
C default values of them, see the block data routine SEBLDAX.
C
        COMMON /SECOMI/ IERRU,IERRF,IRECF,LOMSG
        SAVE   /SECOMI/
C
        COMMON /SECOMC/ ERMSG
          CHARACTER*256 ERMSG
        SAVE   /SECOMC/
C
C Check for an illegal value of IRNEW.
C
        IF (IRNEW.LT.0.OR.IRNEW.GT.2)
     +     CALL SETER ('ENTSR - ILLEGAL VALUE OF THE RECOVERY FLAG',1,2)
C
C Return the previous value of IRECF in IROLD and, if IRNEW is nonzero,
C reset IRECF to that value.
C
        IROLD=IRECF
        IF (IRNEW.NE.0) IRECF=IRNEW
C
C Check for an uncleared prior error that is now unrecoverable.
C
        IF (IERRF.NE.0.AND.IRECF.EQ.2)
     +       CALL SETER ('ENTSR - PRIOR ERROR IS NOW UNRECOVERABLE',2,2)
C
C Done.
C
        RETURN
C
      END

File: ncarg2d/man/error_handling/entsr.m
.TH ENTSR 3NCARG "March 1994" UNIX "NCAR GRAPHICS"
.na
.nh
.SH NAME
ENTSR - Called by a user to set recovery mode in NCAR Graphics.
.SH SYNOPSIS
CALL ENTSR(IROLD,IRNEW)
.SH C-BINDING SYNOPSIS
#include <ncarg/ncargC.h>
.sp
void c_entsr(int *irold, int irnew)
.SH DESCRIPTION 
The FORTRAN statement "CALL ENTSR(IROLD,IRNEW)" is normally used to enter
recovery mode and save the previous value of the internal error-recovery
flag, but it can also be used to exit from recovery mode and save the
previous value of the flag, or to just get the value of the flag, without
changing it.
.sp
If recovery mode is turned off by a call to ENTSR at a time when the
internal error flag is non-zero, this is treated as a fatal error; the
error message is printed, the dump routine FDUM is called, and a STOP
is executed.
.sp
The arguments of ENTSR are as follows:
.sp
.IP "IROLD" 12
(an output variable of type INTEGER) - Receives the old value of the internal
flag that indicates whether recovery mode is in effect or not.  In the former
case, the returned value will be a 1; in the latter case, it will be a 2.
Normally, the value returned is saved for a later call to RETSR.
.sp
.IP "IRNEW" 12
(an input expression of type INTEGER) - Specifies what is to be done to the
internal flag that indicates whether recovery mode is in effect or not.  If
IRNEW is a 1 or a 2, it becomes the new value of the internal recovery-mode
flag: the value 1 turns recovery mode on and the value 2 turns it off.  Other
non-zero values are illegal, but, if IRNEW is zero, the state of the internal
recovery-mode flag is not changed.
.SH C-BINDING DESCRIPTION 
The C-binding argument descriptions are the same as the FORTRAN 
argument descriptions.
.SH EXAMPLES
Use the ncargex command to see the following relevant
examples: 
tseter,
arex02.
.SH ACCESS
To use ENTSR or c_entsr, load the NCAR Graphics libraries ncarg, ncarg_gks,
and ncarg_c, preferably in that order.
.SH SEE ALSO
Online:
eprin, errof, error_handling, fdum, icfell, icloem, nerro, retsr, semess, seter,
ncarg_cbind
.SH COPYRIGHT
Copyright (C) 1987-2009
.br
University Corporation for Atmospheric Research
.br
The use of this Software is governed by a License Agreement.
